results/cmd/api/main.go

```diff
@@ -57,7 +65,44 @@ type server struct {
 	db *sql.DB
 }
 
+// Receive TaskRun from Watcher and save it to local Sqlite Server
 func (s *server) CreateTaskRun(ctx context.Context, req *pb.CreateTaskRunRequest) (*pb.TaskRun, error) {
-	// TODO: implement this.
-	return nil, status.Errorf(codes.Unimplemented, "method CreateTaskRun not implemented")
+	database := s.db
+	statement, err := database.Prepare("INSERT INTO taskrun (taskrunlog, uid, name, namespace) VALUES (?, ?, ?, ?)")
+	if err != nil {
+		log.Fatalf("failed to insert a new taskrun: %v", err)
+	}
+
+	// serialize data and insert it into database
+	taskrunFromClient := req.GetTaskRun()
+	blobData, err := proto.Marshal(taskrunFromClient)
+	if err != nil {
+		log.Fatal("marshaling error: ", err)
+	}
+	if err = proto.Unmarshal(blobData, taskrunFromClient); err != nil {
+		log.Fatal("unmarshaling error: ", err)
+	}
+	taskrunMeta := taskrunFromClient.GetMetadata()
+	if _, err = statement.Exec(blobData, taskrunMeta.GetUid(), taskrunMeta.GetName(), taskrunMeta.GetNamespace()); err != nil {
+		log.Fatalf("failed to execute insertion of a new taskrun: %v", err)
+
+	}
+
+	// check some database information
+	if _, err := database.Query("SELECT logid, taskrunlog FROM taskrun"); err != nil {
+		log.Fatalf("failed to query on database: %v", err)
+	}
+	// var logid int
+	// var taskrunblob []byte
+	// for rows.Next() {
+	// 	rows.Scan(&logid, &taskrunblob)
+	// 	taskrun := &pb.TaskRun{}
+	// 	err = proto.Unmarshal(taskrunblob, taskrun)
+	// 	if err != nil {
+	// 		log.Fatal("unmarshaling error: ", err)
+	// 	}
+	// 	metadata := taskrun.GetMetadata()
+	// 	fmt.Println(strconv.Itoa(logid) + metadata.GetUid() + metadata.GetName() + metadata.GetNamespace())
+	// }
```

Comment:
Let's remove this if it isn't being used.

Reply:
done